analyses/management/commands/merge_mgys_duplicates.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from django.core.management.base import BaseCommand
+from django.db.models import Count
+
+from analyses.models import Study, Assembly, Run, Analysis
+
+
+class Command(BaseCommand):
+    help = "Identify duplicate MGnify studies with the same ENA study and reassign runs and assemblies."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+        )
+
+    def handle(self, *args, **options):
+        self.dry_run = options["dry_run"]
+        self.print_duplicate_mgys_studies()
+
+    #   find ENA studies with two MGYS accessions
+    def print_duplicate_mgys_studies(self):
+        dup_ena_study_ids = (
+            Study.objects.values("ena_study")
+            .annotate(study_count=Count("accession"))
+            .filter(study_count=2)
+            .values_list("ena_study", flat=True)
+        )
+
+        for ena_id in dup_ena_study_ids:
+            logging.info(
+                f"ENA accession {ena_id} is linked to multiple MGnify Studies:"
+            )
+            mgnify_studies = Study.objects.filter(ena_study_id=ena_id).order_by(
+                "accession"
+            )
+
+            for mgys in mgnify_studies:
+                logging.info(f"{mgys.accession}")
+
+            old_study, new_study = mgnify_studies[0], mgnify_studies[1]
+            self.reassign_runs_and_assemblies(old_study, new_study)
+
+    def reassign_runs_and_assemblies(self, old_study, new_study):
+        """
+        Move runs and assemblies from new_study to old_study.
+        Delete the new_study if it's empty of runs, assemblies, and analyses.
+        """
+        # for those edge cases where both studies have the same runs, don't do anything
+        new_runs = Run.objects.filter(study=new_study)
+        old_runs = Run.objects.filter(study=old_study)
+        old_run_accessions = set(a for obj in old_runs for a in obj.ena_accessions)
+        new_run_accessions = set(a for obj in new_runs for a in obj.ena_accessions)
+        if old_run_accessions & new_run_accessions:
+            logging.info(
+                f"DUPLICATE RUNS FOUND IN BOTH STUDIES: old {old_study} and new {new_study}. No further action performed."
+            )
+            return
+
+        new_assemblies = Assembly.objects.filter(assembly_study=new_study)
```

Comment:
This one does indeed make sense to look up this way since there are two reverse relation from assembly to study, depending on whether it is reads study or assembly study 👍